static_token.go

```diff
@@ -1,37 +0,0 @@
-package scalingo
-
-import "context"
-
-// StaticTokenGenerator is an implementation of TokenGenerator which always return the same token.
-// This token is provided to the constructor. The TokenGenerator is used by the Client to
-// authenticate to the Scalingo API.
-//
-// Usage:
-//		t := GetStaticTokenGenerator("my-token")
-//		client := NewClient(ClientConfig{
-//			TokenGenerator: t,
-//		})
-//
-// Any subsequent calls to the Scalingo API will use this token to authenticate.
-type StaticTokenGenerator struct {
-	token  string
-	client *Client
-}
-
-// NewStaticTokenGenerator returns a new StaticTokenGenerator. The only argument is the token that
-// will always be returned by this generator.
-func NewStaticTokenGenerator(token string) *StaticTokenGenerator {
-	return &StaticTokenGenerator{
-		token: token,
-	}
-}
-
-// GetAccessToken always returns the configured token.
-func (t *StaticTokenGenerator) GetAccessToken(context.Context) (string, error) {
-	return t.token, nil
-}
-
-// SetClient sets the client attribute of this token generator.
-func (t *StaticTokenGenerator) SetClient(c *Client) {
-	t.client = c
-}
```

Comment:
I would be in favor of keeping it.
This is still used in internal usage and mocks.

Reply:
OK I reverted the commit and added a comment to explain it's for internal use